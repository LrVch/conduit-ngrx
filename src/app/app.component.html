<div [class]="'theme-wrapper ' + (effectiveTheme$ | async)"
     [ngClass]="{'sticky-header': (stickyHeader$ | async)}">
  <mat-sidenav-container>
    <mat-sidenav [opened]="sideNavOpen$ | async"
                 (closed)="onCloseSideNave()"
                 [mode]="asideOpenMode$ | async">
      <app-aside *ngIf="(settings$ | async)"
                 [theme]="theme$ | async"
                 [themes]="themes$ | async"
                 [stickyHeader]="stickyHeader$ | async"
                 [routeAnimationsEnabled]="routeAnimationsEnabled$ | async"
                 [asideOpenMode]="asideOpenMode$ | async"
                 [asideOpenModes]="asideOpenModes$ | async"
                 [autoNightMode]="autoNightMode$ | async"
                 [autoNightModeFrom]="autoNightModeFrom$ | async"
                 [autoNightModeTo]="autoNightModeTo$ | async"
                 [routeAnimateonsChangeType]="routeAnimateonsChangeType$ | async"
                 [routeAnimateonsChangeTypes]="routeAnimateonsChangeTypes$ | async"
                 (changeRouteAnimationsType)="onChangeRouteAnimationsType($event)"
                 (changeStickyHeader)="onChangeStickyHeader($event)"
                 (changeAsideOpenMode)="onChangeAsideOpenMode($event)"
                 (changeAutoNightMode)="onChangeAutoNightMode($event)"
                 (changeAutoNightModeFrom)="onChangeAutoNightModeFrom($event)"
                 (changeAutoNightModeTo)="onCangeAutoNightModeTo($event)"
                 (changeRouteAnimationsEnable)="onChangeRouteAnimationsEnable($event)"
                 (changeTheme)="onChangeTheme($event)"></app-aside>
    </mat-sidenav>
    <mat-sidenav-content>

      <div class="layout">
        <div class="main">
          <div class="main-content">
            <app-header [user]="user$ | async"
                        [loggedIn]="loggedIn$ | async"
                        [showLoader]="showMailLoader$ | async"
                        [languages]="languages$ | async"
                        [language]="language$ | async"
                        (changeLanguage)="onChangeLanguage($event)"
                        (toggleSideNave)="onToggleSideNave()"></app-header>

            <div [@routeAnimation]="prepRouteState(routerOutlet)">
              <router-outlet #routerOutlet="outlet"></router-outlet>
            </div>
          </div>

          <app-footer></app-footer>
        </div>
      </div>
    </mat-sidenav-content>

  </mat-sidenav-container>

</div>