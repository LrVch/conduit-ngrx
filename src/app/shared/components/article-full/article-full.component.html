<div class="banner">
  <div class="container">
    <h1>{{ article.title }}</h1>
    <h3>{{ article.description }}</h3>

    <br>
    <app-article-user [article]="article"
                      [canModify]="canModify"
                      [locale]="locale"
                      [isDeletingArticle]="isDeletingArticle"
                      (folowing)="onToggleFollowing($event)"
                      (favorited)="onFavoriveToggle()"
                      (delete)="onDeleteArticle()"></app-article-user>

    <app-list-error [errors]="articleErrors"></app-list-error>
  </div>

</div>
<!-- article body -->
<div class="container">
  <div class="row article-full__content">
    <div class="col-md-12">
      <br>
      <br>
      <app-article-body [body]="article.body"></app-article-body>

      <app-tag-list [tagList]="article.tagList"></app-tag-list>
    </div>
  </div>
</div>

<div class="container">
  <div class="row">
    <div class="col-md-12">
      <br>
      <br>
      <app-article-user [article]="article"
                        [canModify]="canModify"
                        [locale]="locale"
                        [isDeletingArticle]="isDeletingArticle"
                        (folowing)="onToggleFollowing($event)"
                        (favorited)="onFavoriveToggle()"
                        (delete)="onDeleteArticle()"></app-article-user>

      <app-list-error [errors]="articleErrors"></app-list-error>
    </div>
  </div>
</div>

<hr />