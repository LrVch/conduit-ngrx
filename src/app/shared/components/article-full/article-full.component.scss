.banner {
    background-color: transparent;
    color: #212529;
    border-bottom: 1px solid #ccc;
}

.article-full__content {
    img {
        max-width: 100%;
    }
}