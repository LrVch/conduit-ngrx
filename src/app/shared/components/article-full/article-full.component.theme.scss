@import '~@angular/material/theming';

@mixin conduit-article-full-component-theme($theme) {
  $primary: map-get($theme, primary);
  $accent: map-get($theme, accent);
  $foreground: map-get($theme, foreground);
  $background: map-get($theme, background);

  app-article-full {
    .banner {
      color: mat-color($foreground, text);
    }
  }
}
