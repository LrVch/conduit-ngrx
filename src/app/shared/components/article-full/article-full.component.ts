import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Article, Errors } from '@app/core';
import { Profile } from 'selenium-webdriver/firefox';

@Component({
  selector: 'app-article-full',
  templateUrl: './article-full.component.html',
  styleUrls: ['./article-full.component.scss']
})
export class ArticleFullComponent {
  @Input('locale') locale: string;
  @Input('article') article: Article = {} as Article;
  @Input('canModify') canModify = false;
  @Input('isDeletingArticle') isDeletingArticle = false;
  @Input('articleErrors') articleErrors: Errors = {} as Errors;
  @Output() favorited = new EventEmitter<Article>();
  @Output() folowing = new EventEmitter<Profile>();
  @Output() delete = new EventEmitter<Article>();


  onToggleFollowing(profile: Profile): void {
    this.folowing.emit(profile);
  }

  onFavoriveToggle(): void {
    this.favorited.emit(this.article);
  }

  onDeleteArticle(): void {
    this.delete.emit(this.article);
  }
}
