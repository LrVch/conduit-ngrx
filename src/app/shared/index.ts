export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './shared.module';
export * from './utils';
export * from './list-error/list-error.component';
export * from './show-authed.directive';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
