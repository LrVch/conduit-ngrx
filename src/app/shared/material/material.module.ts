import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
