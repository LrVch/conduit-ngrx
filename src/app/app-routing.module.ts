import { NgModule } from '@angular/core';
import { Routes, RouterModule, PreloadAllModules } from '@angular/router';
import { PageNotFoundComponent } from './shared';
import { AuthGuard } from './core';
const routes: Routes = [
  {
    path: 'settings',
    loadChildren: './settings/settings.module#SettingsModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'profile',
    loadChildren: './profile/profile.module#ProfileModule',
  },
  {
    path: 'editor',
    loadChildren: './editor/editor.module#EditorModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'article',
    loadChildren: './article/article.module#ArticleModule',
  },
  {
    path: '**',
    component: PageNotFoundComponent,
    data: { title: 'conduit.menu.notfound' }
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes, {
    scrollPositionRestoration: 'enabled'
    // preload all modules; optionally we could
    // implement a custom preloading strategy for just some
    // of the modules (PRs welcome ðŸ˜‰)
    // preloadingStrategy: PreloadAllModules
  })],
  exports: [RouterModule]
})
export class AppRoutingModule { }
