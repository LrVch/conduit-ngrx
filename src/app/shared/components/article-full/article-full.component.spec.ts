import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from '../../../lib/testing';
import { ArticleFullComponent } from './article-full.component';
import { DebugElement, Component, Input, Output, EventEmitter } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '../../material/material.module';
import { Article, Profile } from 'src/app/core';
import { getArticle } from 'src/app/lib/testing/mock-data.helpers';

@Component({
    selector: 'app-article-user',
    template: `
    <a id="follow" (click)="onFollowedToggle()"></a>
    <a id="favorite" (click)="onFavoriteToggle()"></a>
    <a id="delete" (click)="onDeleteToggle()"></a>
    `
})
class TestHostUserComponent {
    @Input('canModify') canModify: boolean;
    @Input('article') article: Article;
    @Input('isDeletingArticle') isDeletingArticle: boolean;
    @Output() folowing = new EventEmitter<Profile>();
    @Output() favorited = new EventEmitter<Article>();
    @Output() delete = new EventEmitter<any>();

    onFollowedToggle(): void {
        this.folowing.emit(this.article.author);
    }
    onFavoriteToggle(): void {
        this.favorited.emit(this.article);
    }
    onDeleteToggle(): void {
        this.delete.emit();
    }
    constructor() { }
}

@Component({
    selector: 'app-article-body',
    template: ``
})
class TestHostBodyComponent {
    @Input('body') body: string;
}

@Component({
    selector: 'app-list-error',
    template: ``
})
class TestHostErrorsComponent {
    @Input('errors') errors: any;
}

@Component({
    selector: 'app-tag-list',
    template: ``
})
class TestHostTagsComponent {
    @Input('tagList') tagList: any;
}

describe('ArticleFullComponent', () => {
    let component: ArticleFullComponent;
    let fixture: ComponentFixture<ArticleFullComponent>;
    let de: DebugElement;
    let el: HTMLElement;
    const initialArticle: Article = {
        author: {},
        tagList: []
    } as Article;

    const article: Article = getArticle();

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [
                        ArticleFullComponent,
                        TestHostBodyComponent,
                        TestHostUserComponent,
                        TestHostErrorsComponent,
                        TestHostTagsComponent
                    ],
                    imports: [MaterialModule]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(ArticleFullComponent);
                component = fixture.componentInstance;
                component.article = initialArticle;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has "article" @Input', () => {
        expect(component.article).toEqual(initialArticle);
    });

    it('should has "canModify" @Input', () => {
        expect(component.canModify).toEqual(false);
    });

    it('should has "isDeletingArticle" @Input', () => {
        expect(component.canModify).toEqual(false);
    });

    it('should has "articleErrors" @Input', () => {
        expect(component.articleErrors).toEqual({});
    });

    it('should show "article" @Input', () => {
        component.article = getArticle();
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should raise favorited event and pass an selected article', () => {
        let selectedArticle;
        component.article = article;
        component.favorited.subscribe(a => selectedArticle = a);

        fixture.detectChanges();

        const favoriteDe = de.query(By.css('app-article-user')).query(By.css('#favorite'));

        favoriteDe.triggerEventHandler('click', null);
        expect(selectedArticle).toEqual(article);
    });

    it('should raise folowing event and pass an selected profile', () => {
        let selectedProfile;
        component.article = article;

        component.folowing.subscribe(a => selectedProfile = a);

        fixture.detectChanges();

        const followDe = de.query(By.css('app-article-user')).query(By.css('#follow'));

        followDe.triggerEventHandler('click', null);
        expect(selectedProfile).toEqual(article.author);
    });

    it('should raise delete event', () => {
        let deletedArticle;
        component.article = article;
        component.canModify = true;

        component.delete.subscribe(a => deletedArticle = a);

        fixture.detectChanges();

        const deleteDe = de.query(By.css('app-article-user')).query(By.css('#delete'));

        deleteDe.triggerEventHandler('click', null);

        expect(deletedArticle).toEqual(article);
    });
});

