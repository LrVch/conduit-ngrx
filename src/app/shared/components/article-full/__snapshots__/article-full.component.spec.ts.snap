// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleFullComponent should show "article" @Input 1`] = `
<app-article-full
  article={[Function Object]}
  articleErrors={[Function Object]}
  canModify="false"
  delete={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  folowing={[Function EventEmitter]}
  isDeletingArticle="false"
>
  <div
    class="banner"
  >
    <div
      class="container"
    >
      <h1>
        title
      </h1>
      <h3>
        desc
      </h3>
      <br />
      <app-article-user
        ng-reflect-article="[object Object]"
        ng-reflect-can-modify="false"
        ng-reflect-is-deleting-article="false"
      >
        <a
          id="follow"
        />
        <a
          id="favorite"
        />
        <a
          id="delete"
        />
      </app-article-user>
      <app-list-error
        ng-reflect-errors="[object Object]"
      />
    </div>
  </div><div
    class="container"
  >
    <div
      class="row article-full__content"
    >
      <div
        class="col-md-12"
      >
        <br />
        <br />
        <app-article-body
          ng-reflect-body="body"
        />
        <app-tag-list
          ng-reflect-tag-list="list"
        />
      </div>
    </div>
  </div><div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-12"
      >
        <br />
        <br />
        <app-article-user
          ng-reflect-article="[object Object]"
          ng-reflect-can-modify="false"
          ng-reflect-is-deleting-article="false"
        >
          <a
            id="follow"
          />
          <a
            id="favorite"
          />
          <a
            id="delete"
          />
        </app-article-user>
        <app-list-error
          ng-reflect-errors="[object Object]"
        />
        <app-list-error
          ng-reflect-errors="[object Object]"
        />
      </div>
    </div>
  </div><hr />
</app-article-full>
`;
